From: Tomas Mraz <tmraz@fedoraproject.org>
Date: Wed, 4 Nov 2020 09:37:36 +0100
Subject: pam_env: allow environment files without EOL at EOF

Fixes #263

* modules/pam_env/pam_env.c (_assemble_line): Do not error out if at feof()

Origin: upstream, https://github.com/linux-pam/linux-pam/commit/12824dd648b0668968231044ed805d1f3b212d7e
Applied-Upstream: 1.5.0
Bug: https://github.com/linux-pam/linux-pam/issues/263
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/pam/+bug/1953201
Reviewed-By: Sergio Durigan Junior <sergiodj@ubuntu.com>
---
 modules/pam_env/pam_env.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/pam_env/pam_env.c b/modules/pam_env/pam_env.c
index 79d4372..4aa3777 100644
--- a/modules/pam_env/pam_env.c
+++ b/modules/pam_env/pam_env.c
@@ -311,7 +311,7 @@ static int _assemble_line(FILE *f, char *buffer, int buf_len)
 	    D(("_assemble_line: corrupted or binary file"));
 	    return -1;
 	}
-	if (p[strlen(p)-1] != '\n') {
+	if (p[strlen(p)-1] != '\n' && !feof(f)) {
 	    D(("_assemble_line: line too long"));
 	    return -1;
 	}
